src/libYARP_dev/src/devices/AnalogWrapper/AnalogWrapper.cpp

```diff
@@ -457,8 +481,53 @@ bool AnalogWrapper::checkROSParams(Searchable &config)
             useROS = ROS_config_error;
             return false;
         }
-        frame_id = rosGroup.find("frame_id").asString();
-        yInfo() << sensorId << "frame_id is " << frame_id;
+
+        if(rosGroup.find("frame_id").isString())
+        {
+            frame_idVec.at(0) = rosGroup.find("frame_id").asString();
+            yInfo() << sensorId << "frame_id is " << frame_idVec.at(0);
+
+        }
+        else if(rosGroup.find("frame_id").isList())
+        {
+            yarp::os::Bottle *frame_idBottle = rosGroup.find("frame_id").asList();
+
+            if(frame_idBottle->size() != rosTopicNamesVec.size())
+            {
+                yError("AnalogWrapper: mismatch between the number of ros topics and frame_ids");
+                return false;
+            }
+
+            yInfo() << sensorId << "frame_id list is " << frame_idBottle->toString();
+
+            frame_idVec.resize(frame_idBottle->size());
+            for(size_t i = 0; i < frame_idBottle->size(); i++)
+            {
+                frame_idVec.at(i) = frame_idBottle->get(i).asString();
+            }
+        }
+
+        if(!rosGroup.check("rosOffset"))
+        {
+            yWarning() << sensorId << "cannot find rosOffset parameter, using the default offset 0 while reading analog sensor data";
+        }
+
+        if(rosGroup.find("rosOffset").isInt())
```

Comment:
`isInt` and `asInt` are no longer allowed in YARP (they print deprecation warnings), please use `isInt32` and `asInt32` next time.